Batteries/Data/List/Lemmas.lean

```diff
@@ -19,6 +19,15 @@ namespace List
 
 @[deprecated (since := "2024-08-15")] alias isEmpty_iff_eq_nil := isEmpty_iff
 
+/-! ### head and tail -/
+
+theorem head_cons_tail : ∀ l h, @head α l h :: l.tail = l
```

Comment:
Can you write this without the `@`?

Reply:
In the following code, the `@` symbol was used in a similar way: https://github.com/leanprover-community/batteries/blob/5f963d5d06cc3c2d3abd0806891133137a59d7eb/Batteries/Data/List/Lemmas.lean#L22-L25.

Reply:
Using `@` is a bit old fashioned. This would math the style of the next theorem:
```suggestion
theorem head_cons_tail : ∀ (l : List α) (hne : l ≠ []), l.head hne :: l.tail = l
```